### PR TITLE
Subscriptions plugins

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Adds a plugin for subscriptions with a rabbitmq broker

### CODE REVIEW

File: packages/plugins/graphql-subscriptions-rabbitmq/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        "supertest": "6.2.3",
+        "ts-jest": "27.1.4",
+        "typescript": "4.6.3",
+        "ws": "8.5.0"
+    },
+    "dependencies": {
+        "amqplib": "^0.8.0"
+    },
+    "peerDependencies": {
+        "@neo4j/graphql": "^3.1.0"
```
Comment (angrykoala) [inline packages/plugins/graphql-subscriptions-rabbitmq/package.json:55]: not sure if this is the best approach to share types :thinking:

File: packages/plugins/graphql-subscriptions-rabbitmq/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import amqp from "amqplib";
+import { EventEmitter } from "events";
+import { Neo4jGraphQLSubscriptionsPlugin, SubscriptionsEvent } from "@neo4j/graphql";
+import { AmqpApi, ConnectionOptions } from "./amqp-api";
+
+export { ConnectionOptions } from "./amqp-api";
+
+const DEFAULT_EXCHANGE = "neo4j-graphql";
+
+export type Neo4jGraphQLSubscriptionsRabbitMQContructorOptions = { exchange?: string };
```
Comment (tbwiss) [inline packages/plugins/graphql-subscriptions-rabbitmq/src/index.ts:29]:
```suggestion
export type Neo4jGraphQLSubscriptionsRabbitMQConstructorOptions = { exchange?: string };
```

**Review by tbwiss — COMMENTED** [review]
Good one!

File: config/release-changelog-builder-action/configuration-plugin-auth copy.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply config/release-changelog-builder-action/configuration-plugin-auth copy.json:1]: @darrellwarde

File: packages/graphql/tests/e2e/setup/neo4j.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as neo4j from "neo4j-driver";
+import * as util from "util";
+
+let driver: neo4j.Driver;
+
+async function connect(): Promise<neo4j.Driver> {
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/tests/e2e/setup/neo4j.ts:25]: Using the setup files from a different suite seems like an antipattern. We could have moved this to some "common" setup, but tbh this setup is just coincidentally the same (for now)

File: packages/plugins/graphql-subscriptions-rabbitmq/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# @neo4j/graphql-plugin-subscriptions-rabbitmq
+
+Subscription plugin with RabbitMQ broker for @neo4j/graphql plugins for `@neo4j/graphql`
+
+1. [Documentation](https://neo4j.com/docs/graphql-manual/current/auth/)
+
+## Installation
+
+```
+$ npm install @neo4j/graphql-plugin-auth
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/plugins/graphql-subscriptions-rabbitmq/README.md:10]: Good catch

File: packages/plugins/graphql-subscriptions-rabbitmq/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+export class Neo4jGraphQLSubscriptionsRabbitMQ implements Neo4jGraphQLSubscriptionsPlugin {
+    public events: EventEmitter;
+    private amqpApi: AmqpApi<SubscriptionsEvent>;
+
+    constructor(options: Neo4jGraphQLSubscriptionsRabbitMQContructorOptions = {}) {
```
Comment (angrykoala) [inline packages/plugins/graphql-subscriptions-rabbitmq/src/index.ts:35]:
```suggestion
    constructor(options: Neo4jGraphQLSubscriptionsRabbitMQConstructorOptions = {}) {
```

**Pull request merged**